tests/_async/test_http2.py

```diff
@@ -83,6 +83,9 @@ async def test_http2_connection_closed():
         with pytest.raises(RemoteProtocolError):
             await conn.request("GET", "https://example.com/")
 
+        with pytest.raises(ConnectionNotAvailable):
+            await conn.request("GET", "https://example.com/")
```

Comment:
This doesn't seem right as we still raise a `RemoteProtocolError` on the first failure